### PR TITLE
CachedPage: move to a proper file, add tests

### PR DESCRIPTION
**Pull request by virtualtam**
Modifications
 - rename `pageCache` to `CachedPage`
 - move utilities to `Cache`
 - do not access globals
 - apply coding rules
 - update LinkDB and test code
 - add test coverage

### CODE REVIEW

File: application/PageCache.php [hunk 1]
```diff
@@ -0,0 +1,105 @@
...
+}
+
+
+/**
+ * Simple cache system, mainly for the RSS/ATOM feeds
+ */
+class PageCache
+{
+    // Directory containing page caches
+    private $_cacheDir;
```
Comment (ArthurHoaro) [inline application/PageCache.php]: I thought we didn't use underscore for private variables?

Comment (virtualtam) [inline application/PageCache.php]: Hence the big caps "[WIP]" :)

Still needs some work (the PR is mostly for issue tracking) => movin' to the 0.7.0 milestone

File: application/Cache.php [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+    // TODO: check write access to the cache directory
+
+    $handler = opendir($pageCacheDir);
+    if ($handler == false) {
+        return;
+    }
+
+    while (($filename = readdir($handler)) !== false) {
+        if (endsWith($filename, '.cache')) {
+                unlink($pageCacheDir.'/'.$filename);
```
Comment (nodiscc) [inline application/Cache.php]: Will this fail silently? `$pageCacheDir` could be writable but not some `.cache` file inside (wrong permissions)

Comment (virtualtam) [inline application/Cache.php]: this function may need to be rewritten a bit:
- there should only be `.cache` files in the `pagecache` directory
    - so wiping all content without considering the extension should suffice
- surprisingly, there is no built-in way to recursively remove a directory's content in PHP, see
    - [rmdir](http://php.net/manual/en/function.rmdir.php)
    - [unlink](http://php.net/manual/en/function.unlink.php) (see also the top-voted comment)
    - [php - delete directory with files in it?](http://stackoverflow.com/questions/3349753/delete-directory-with-files-in-it) (SO)
- directory and file permissions need to be handled

exception handling would be appropriate for these cases, brought either in an additional commit or a separate PR

Comment (virtualtam) [inline application/Cache.php]: After a quick test:

``bash
$ sudo touch tests/dummycache/toto.cache
$ ls -l tests/dummycache/
total 0
-rw-r--r-- 1 root root 0  aug 6  18:11 toto.cache
$ make test
-------
PHPUNIT
-------
PHPUnit 4.6.10 by Sebastian Bergmann and contributors.

Configuration read from /home/tamisier/public_html/shaarli/phpunit.xml

EE............................................................... 65 / 65 (100%)


Time: 1.36 seconds, Memory: 10.25Mb

There were 2 errors:

1) CachedTest::testPurgeCachedPages
file_put_contents(tests/dummycache/toto.cache): failed to open stream: Permission denied

/home/virtualtam/public_html/shaarli/tests/CacheTest.php:33

2) CachedTest::testInvalidateCaches
file_put_contents(tests/dummycache/toto.cache): failed to open stream: Permission denied

/home/virtualtam/public_html/shaarli/tests/CacheTest.php:33

FAILURES!
Tests: 65, Assertions: 133, Errors: 2.
```

**Pull request merged**